core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -335,6 +352,52 @@ public boolean isSamplingEnabled() {
     public SamplerConfiguration getSamplerConfiguration() {
       return RFileScanner.this.getSamplerConfiguration();
     }
+
+    @Override
+    public TableId getTableId() {
+      return null;
+    }
+
+    @Override
+    @Deprecated(since = "2.0.0")
+    public AccumuloConfiguration getConfig() {
+      return tableConf;
+    }
+
+    @Override
+    @Deprecated(since = "2.1.0")
+    public ServiceEnvironment getServiceEnv() {
+      return serviceEnvironment.get();
+    }
+
+    private ServiceEnvironment createServiceEnv() {
+      return new ServiceEnvironment() {
+        @Override
+        public <T> T instantiate(TableId tableId, String className, Class<T> base) {
+          throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public <T> T instantiate(String className, Class<T> base) {
+          throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public String getTableName(TableId tableId) {
+          throw new UnsupportedOperationException();
```

Comment:
All of these exceptions could include the same message.  Maybe make the message a private constant.

```suggestion
          throw new UnsupportedOperationException("This scanner is unrelated to any table or accumulo instance, it is operating directly on files. Therefore it can not support this operation.");
```

Reply:
Done